packages/pools/src/router/routes.ts

```diff
@@ -150,6 +150,50 @@ export class OptimizedRoutes implements TokenOutGivenInRouter {
 
     let routes = this.getCandidateRoutes(tokenIn.denom, tokenOutDenom);
 
+    for (let i = 0; i < routes.length; i++) {
```

Comment:
`routes.forEach((route) => { })`

Reply:
actually it looks like maybe you want a `map` here? `map` returns a new array

Reply:
I want to return from the caller function if I successfully find the desired route.

If I use for each, I would be returning from the callback, right?

Reply:
I was unable to get this working because I'm looking to return from `getOptimizedRoutesByTokenIn` (the caller) if certain criteria are matches.

With `map` or `forEach`, I would be returning from the callback.

Additionally, with `map`, it would make it return an array of a different type - `[]RouteWithInAmount`. routes is `[]Route` type. This change would make further handling messier. I think it's best to simply return in a for loop.

Please let me know if that makes sense